databroker/eventsource/archiver.py

```diff
@@ -73,65 +119,155 @@ def descriptors_given_header(self, header):
                         'external_url': self.url}
                 descs.append(desc)
             self._descriptors[run_start_uid] = descs
-            return [ d for d in self._descriptors]
+            
+            return list(self._descriptors[run_start_uid])      
+            # return [ d for d in self._descriptors] 
+            
             #prepare = partial(self.prepare_hook, 'descriptor')
             #return list(map(prepare, self._descriptors[run_start_uid]))
 
-    def docs_given_header(self, header,fill=False, fields=None,
-                          **kwargs):
+    def docs_given_header(self, header, stream_name='ALL', fields=None):
+        """
+        Get documents for given Header.
+
+        Parameters
+        ----------
+        header: Header
+            The header to fetch the events for
+        stream_name: string, not used 
+            names of interest are defined via user-defined PVs
+        fields: list, not used
+            names of interest are defined via user-defined PVs
+
+        Yields
+        ------
+        str: name
+            The name of the document being yielded
+        doc: Document
+            The data payload
+        """
+
         desc_uids = {}
+
+        since, until = header['start']['time'], header['stop']['time']
+        _from = _munge_time(since, self.tz)
+        _to = _munge_time(until, self.tz)
+
+        yield 'start', header['start']
+        
         for d in self.descriptors_given_header(header):
+
+            yield 'descriptor', d
+            
             # Stash the desc uids in a local var so we can use them in events.
+            name = list(d['data_keys'].keys())[0]
             pv = list(d['data_keys'].values())[0]['source']
             desc_uids[pv] = d['uid']
-            yield d
-        since, until = header['start']['time'], header['stop']['time']
-        for name, pv in six.iteritems(self.pvs):
-            _from = _munge_time(since, self.tz)
-            _to = _munge_time(until, self.tz)
+            
             params = {'pv': pv, 'from': _from, 'to': _to}
+            
             req = requests.get(self.archiver_addr, params=params, stream=True)
             req.raise_for_status()
             raw, = req.json()
+            
             timestamps = [x['secs'] for x in raw['data']]
             data = [x['val'] for x in raw['data']]
-            for seq_num, (d, t) in enumerate(zip(data, timestamps)):
-                doc = {'data': {name: d},
+            
+            for seq_num, (v, t) in enumerate(zip(data, timestamps)):
```

Comment:
We start `seq_num` at `1` -- which is unfortunate in the sense that it's not very Python, but at this point we'd better be consistent. You can do `enumerate(..., start=1)` to start counting from `1`.

Reply:
sure